### DISCUSSION TITLE
Calculate Field with Formula Type

### DISCUSSION THREAD

**Discussion by muksidin** [Budibase/budibase#11479] [Help] [answered]
Hello, good day.

I have some columns or field with type formula, ie field `total_lembur` & `kalkulasi` are formula type and . below is code,

```
var hasil = Number($("total_lembur")) * 2;
return Number(hasil);
```

but the result is NAN. 

i already attach the picture

![Screenshot_1](https://github.com/Budibase/budibase/assets/5665115/6c938496-2b19-4ac3-b3d7-4d866756e4dc)

i use MySQL database but if i try using budibase database that formula is okay.

**Comment by ConorWebb96** [comment, marked as answer]
Hey @muksidin,

Its returning NaN because you are trying to do math on a string. You will need to make a few amends to the above JS. Firstly, by splitting the number and text out, then parsing the number into int and from there doing your calculations and then returning everything as a string again.

<img width="404" alt="Screenshot 2023-08-09 at 07 49 07" src="https://github.com/Budibase/budibase/assets/126772285/925d0c4d-7fa3-4305-b54f-0fb53d21636e">

<img width="602" alt="Screenshot 2023-08-09 at 07 52 53" src="https://github.com/Budibase/budibase/assets/126772285/1ddb3d37-c963-47ce-84a1-6a11c4ef9399">

```
const originalString = $("total_lembur");
const match = originalString.match(/(\d+)(\D+)/);

if (match) {
  const numberPart = parseInt(match[1], 10);
  const textPart = match[2];

  const doubledNumber = numberPart * 2;

  const newText = doubledNumber.toString() + ' ' + textPart;
  return newText; 
} else {
  return originalString;
}
```

Note the above will only work if the number is first and the text is second. Otherwise you will need to amend this to include such cases as well.

I hope this helps!

**Reply by muksidin** [reply]
Ok thanks, @ConorWebb96 . Very helpfull 👍
